gradoop-temporal/src/test/java/org/gradoop/temporal/model/impl/operators/keyedgrouping/keys/TimeStampKeyFunctionTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.temporal.model.impl.operators.keyedgrouping.keys;
+
+import org.gradoop.common.model.api.entities.VertexFactory;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.api.functions.KeyFunctionWithDefaultValue;
+import org.gradoop.temporal.model.api.TimeDimension;
+import org.gradoop.temporal.model.impl.operators.keyedgrouping.TemporalGroupingKeys;
+import org.gradoop.temporal.model.impl.pojo.TemporalElement;
+import org.gradoop.temporal.model.impl.pojo.TemporalVertex;
+import org.gradoop.temporal.util.TemporalGradoopTestBase;
+import org.junit.Test;
+
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalField;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Test for the {@link TimeStampKeyFunction} key function.
+ */
+public class TimeStampKeyFunctionTest extends TemporalGradoopTestBase {
```

Comment:
These tests seem like a good fit for parameterization. Maybe we should use TestNG here.

Reply:
Correct, we would have to also use it in other classes of `gradoop-temporal`, I think a follow-up issue would be a good idea here